src/views/Timesheets/index.js

```diff
@@ -0,0 +1,539 @@
+import MuiAlert from '@material-ui/lab/Alert';
+import DateFnsUtils from '@date-io/date-fns';
+import jobs from '../../services/jobs';
+import { MuiPickersUtilsProvider, TimePicker, DatePicker } from '@material-ui/pickers';
+import ShiftDisplay from './components/ShiftDisplay';
+import './timesheets.css';
+
+const Timesheets = (props) => {
+  const [userJobs, setUserJobs] = useState([]);
+  const [selectedDateIn, setSelectedDateIn] = useState(null);
+  const [selectedDateOut, setSelectedDateOut] = useState(null);
+  const [selectedJob, setSelectedJob] = useState(null);
+  const [shiftTooLong, setShiftTooLong] = useState(false);
+  const [shiftTooShort, setShiftTooShort] = useState(false);
+  const [timeOutIsBeforeTimeIn, setTimeOutIsBeforeTimeIn] = useState(false);
+  const [enteredFutureTime, setEnteredFutureTime] = useState(false);
+  const [hoursWorkedInDecimal, setHoursWorkedInDecimal] = useState(0.0);
+  const [userShiftNotes, setUserShiftNotes] = useState('');
+  const [isOverlappingShift, setIsOverlappingShift] = useState(false);
+  const [shiftDisplayComponent, setShiftDisplayComponent] = useState(null);
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+
+  const handleTimeOutIsBeforeTimeIn = (timeIn, timeOut) => {
+    if (timeIn !== null && timeOut !== null) {
+      let timeDiff = timeOut.getTime() - timeIn.getTime();
+      let calculatedTimeDiff = timeDiff / 1000 / 60 / 60;
+      let roundedHourDifference = (Math.round(calculatedTimeDiff * 4) / 4).toFixed(2);
+      setHoursWorkedInDecimal(roundedHourDifference);
+      let hoursWorked = Math.floor(calculatedTimeDiff);
+      let minutesWorked = Math.round((calculatedTimeDiff - hoursWorked) * 60);
+
+      if (minutesWorked >= 60) {
+        hoursWorked++;
+        minutesWorked = 0;
+      }
+
+      if (timeDiff < 0) {
+        setTimeOutIsBeforeTimeIn(true);
+      } else {
+        setTimeOutIsBeforeTimeIn(false);
+
+      }
+      if (calculatedTimeDiff > 20) {
+        setShiftTooLong(true);
+      } else {
+        setShiftTooLong(false);
+      }
+
+      if (calculatedTimeDiff < 0.25) {
+        setShiftTooShort(true);
+      } else {
+        setShiftTooShort(false);
+      }
+    }
+  };
+
+  const checkForFutureDate = () => {
+    let now = Date.now();
+    setEnteredFutureTime((selectedDateIn.getTime() > now) || (selectedDateOut.getTime() > now));
+  }
+
+  if (props.Authentication) {
+    function Alert(props) {
+      return <MuiAlert elevation={6} variant="filled" {...props} />;
+    }
+
+    const getActiveJobsForUser = () => {
+      let details = {
+        shift_start_datetime: selectedDateIn.toLocaleString(),
+        shift_end_datetime: selectedDateOut.toLocaleString(),
+      };
+      jobs.getActiveJobsForUser(details).then(result => {
+        setUserJobs(result);
+      });
+    };
+
+    const getSavedShiftsForUser = () => {
+      return jobs.getSavedShiftsForUser();
+    };
+
+    const handleDateChange1 = date => {
```

Comment:
Would a more fitting name for these (and the one on 103) be handleTimeChangeIn and Out?

Reply:
"this" not "these"

Reply:
Yes, their current names are a relic of the early stages of development. We changed the names of the objects they deal with, but forgot to change the names of the method... whoops